### PR TITLE
refactor func.d: add const

### PR DESCRIPTION
**Pull request by WalterBright**
Progress in using `const`

### CODE REVIEW

File: src/dmd/func.d [hunk 1]
```diff
@@ -1561,21 +1561,26 @@ extern (C++) class FuncDeclaration : Declaration
...
                 printf("\tmember of %s\n", cd.toChars());
-        }
-        return isMember() && (Declaration.isFinal() || ((cd = toParent().isClassDeclaration()) !is null && cd.storage_class & STC.final_));
+        }}
+        if (!isMember())
+            return false;
+        if (Declaration.isFinal())
+            return true;
+        auto cd = toParent().isClassDeclaration();
+        return (cd !is null && cd.storage_class & STC.final_);
```
Comment (marler8997) [inline src/dmd/func.d]: *NIT
unhelpful parens? 
```D
return cd !is null && cd.storage_class & STC.final_;
```
 Maybe you meant to do:
```D
return (cd !is null) && (cd.storage_class & STC.final_);
```

**Review by marler8997 — APPROVED** [review]

**Review by marler8997 — APPROVED** [review]

**Comment by JinShil** [comment]
Doesn't this also need to be reflected in the C++ headers?

**Comment by WalterBright** [comment]
For virtual functions, not really since they are accessed through the vtbl[], but I added them anyway.

**Review by JinShil — APPROVED** [review]

**Comment by JinShil** [comment]
Looks like all tests in Jenkins passed, but the GitHub status didn't update.  Therefore, I'm manually merging.

**Comment by wilzbach** [comment]
> Looks like all tests in Jenkins passed, but the GitHub status didn't update. 

FYI: https://github.com/dlang/ci/issues/133 (the GH API rate limit sometimes gets exceeded, it's probably due to Martin using the same token as it's used for dlang-bot)

**Pull request merged**